### PR TITLE
Add `chr22q` loss variable to ATRT molecular subtyping

### PR DESCRIPTION
**Pull request by cbethell**
### Purpose/implementation Section

The purpose of this PR is to add chr 22 loss information to the ATRT subtype data.frame and tsv file.

#### What scientific question is your analysis addressing?

This analysis is addressing the molecular subtyping of ATRT samples.

#### What was your approach?

I used the `broad_values_by_arm.txt` file from GISTIC to obtain the chr22q data for the ATRT samples.

#### What GitHub issue does your pull request address?

This PR addresses issue #244.

### Directions for reviewers. Tell potential reviewers what kind of feedback you are soliciting.

Does this additional information appear to be accurately represented?

#### Is the analysis in a mature enough form that the resulting figure(s) and/or table(s) are ready for review?

Yes, this is ready for review.

#### What types of results are included (e.g., table, figure)?

This PR includes changes to the heatmap:
- plots/atrt_heatmap.png

This plot can be viewed in the `README` [here](https://github.com/cbethell/OpenPBTA-analysis/blob/add-chr-22-atrt/analyses/molecular-subtyping-ATRT/README.md).

It also includes changes to the final output tsv file:
- results/ATRT_molecular_subtypes.tsv.gz

#### Reproducibility Checklist

- [x] The dependencies required to run the code in this pull request have been added to the project Dockerfile.
- [x] This analysis has been added to continuous integration.
- [x] This analysis is recorded in the table in `analyses/README.md`.

### CODE REVIEW

**Comment by cbethell** [comment]
The html output with the final table displayed can be found [here](https://cbethell.github.io/open-pbta-output/01-ATRT-molecular-subtyping-data-prep.nb.html#gistic_data).

**Comment by jaclyn-taroni** [comment]
It could be useful to organize the table columns such that the relevant information for each of the subtypes on #244 is grouped/presented next to each other. I think we at least want the _SMARCB1_ status and chr22q columns to be grouped and come earlier in the table.

**Comment by cbethell** [comment]
> It could be useful to organize the table columns such that the relevant information for each of the subtypes on #244 is grouped/presented next to each other. I think we at least want the _SMARCB1_ status and chr22q columns to be grouped and come earlier (further to the left) in the table.

View the table reflecting the suggestion above [here](https://cbethell.github.io/open-pbta-output/01-ATRT-molecular-subtyping-data-prep.nb.html#save_final_table_of_results).

**Comment by jaclyn-taroni** [comment]
@cbethell I noticed a couple instances where GISTIC data is missing but the focal data is not and vice versa (here are two examples): 

| sample_id | 	Kids_First_Biospecimen_ID |	Kids_First_Participant_ID | chr_22q_loss |	SMARCB1_focal_status |
|----|----|----|----|----|
| 7316-376 | BS_74A1TB03, BS_M4923M40 |	PT_MTE126WM | NA | loss, neutral |
| 7316-2187 | BS_53TV75NN, BS_850BAHH9 | PT_HVZTF42R | No | NA |

Do you know why that might be?

**Comment by cbethell** [comment]
> @cbethell I noticed a couple instances where GISTIC data is missing but the focal data is not and vice versa (here are two examples):
> 
> sample_id	Kids_First_Biospecimen_ID	Kids_First_Participant_ID	chr_22q_loss	SMARCB1_focal_status
> 7316-376	BS_74A1TB03, BS_M4923M40	PT_MTE126WM	NA	loss, neutral
> 7316-2187	BS_53TV75NN, BS_850BAHH9	PT_HVZTF42R	No	NA
> Do you know why that might be?

@jaclyn-taroni  the `NA`'s are due to data missing for these samples from the gistic data and focal data, respectively.

**Comment by jaclyn-taroni** [comment]
The GISTIC data and focal CN data you are using are both derived from the CNVkit data. That suggests to me that something else might be going on. Are all the samples that end up in that table (with WGS data) in the CNVkit file?

**Comment by cbethell** [comment]
@jaclyn-taroni all of the samples are in the CNVkit file, but they do not all have data for `SMARCB1`. These samples are however not all in the GISTIC file.

**Comment by jaclyn-taroni** [comment]
> all of the samples are in the CNVkit file, but they do not all have data for `SMARCB1`.
Ah, okay - so this suggests to me that these should be set as `neutral` or something else rather than `NA` because it's not that the data is missing but that we don't have evidence that it's a loss - do you agree?

**Comment by cbethell** [comment]
> Ah, okay - so this suggests to me that these should be set as `neutral` or something else rather than `NA` because it's not that the data is missing but that we don't have evidence that it's a loss - do you agree?

Yes, that is a good point. I will implement this change now.

**Comment by jaclyn-taroni** [comment]
@cbethell can you generate a list of `Kids_First_Biospecimen_ID` that are in the CNVkit data but not in the GISTIC data and [file a data issue](https://github.com/AlexsLemonade/OpenPBTA-analysis/issues/new?assignees=&labels=data&template=data-question.md&title=) please?

**Comment by cbethell** [comment]
> @cbethell can you generate a list of `Kids_First_Biospecimen_ID` that are in the CNVkit data but not in the GISTIC data and [file a data issue](https://github.com/AlexsLemonade/OpenPBTA-analysis/issues/new?assignees=&labels=data&template=data-question.md&title=) please?

@jaclyn-taroni yes, I'm on it 👍

**Comment by jharenza** [comment]
> > @cbethell can you generate a list of `Kids_First_Biospecimen_ID` that are in the CNVkit data but not in the GISTIC data and [file a data issue](https://github.com/AlexsLemonade/OpenPBTA-analysis/issues/new?assignees=&labels=data&template=data-question.md&title=) please?
> 
> @jaclyn-taroni yes, I'm on it 👍

Hi! Regarding this - I checked the `sample_seg_counts.txt` file that prints from the GISTIC run and found that the one sample that had `NA` in GISTIC but had CNVkit calls, `BS_74A1TB03`, actually had too many segments for GISTIC (>2500) for calls to be made and was excluded from GISTIC analyses. The `sample_seg_counts.txt` file has all sample segment counts and whether they were included or excluded, so that explains the discrepancies.

**Comment by jaclyn-taroni** [comment]
@jharenza we figured that there was some GISTIC filtering step. We do not have much documentation around the GISTIC files at the moment: https://github.com/AlexsLemonade/OpenPBTA-analysis/blob/master/doc/data-formats.md#copy-number-files

**Comment by jharenza** [comment]
OK - I think I will have to update that.

**Comment by cbethell** [comment]
> Hi! Regarding this - I checked the `sample_seg_counts.txt` file that prints from the GISTIC run and found that the one sample that had `NA` in GISTIC but had CNVkit calls, `BS_74A1TB03`, actually had too many segments for GISTIC (>2500) for calls to be made and was excluded from GISTIC analyses. The `sample_seg_counts.txt` file has all sample segment counts and whether they were included or excluded, so that explains the discrepancies.

@jharenza thank you for this clarification. It indeed does explain the discrepancies as I cross checked said samples in a separate notebook [here](https://cbethell.github.io/open-pbta-output/GISTIC-missing-samples.nb.html#missing_samples).

I have also removed the duplicated columns in the final tsv file.

**Comment by jaclyn-taroni** [comment]
My review of #410 made me think that a similar issue might be happening here with the CNVkit data. I have a few changes that I have not yet pushed because they conflict with the last commit and I should take another look. Because of how we filtered the subset files initially, we were inadvertently dropping a few samples without transcriptomic data so the table will have some additional samples but largely remain the same.

File: analyses/molecular-subtyping-ATRT/00-subset-files-for-ATRT.R [hunk 1]
```diff
@@ -81,26 +81,31 @@ tmb_df <-
...
+# Read in GISTIC `broad_values_by_arm.txt` file
+gistic_df <-
+  data.table::fread(unzip(
+    file.path(root_dir, "data", "pbta-cnv-cnvkit-gistic.zip"),
+    files = file.path(
+      "2019-12-10-gistic-results-cnvkit",
+      "broad_values_by_arm.txt"
+    ),
+    exdir = file.path(root_dir, "scratch")
+  ))
```
Comment (jaclyn-taroni) [inline analyses/molecular-subtyping-ATRT/00-subset-files-for-ATRT.R]:
```suggestion
  ), data.table = FALSE)
```

File: analyses/molecular-subtyping-ATRT/00-subset-files-for-ATRT.R [hunk 2]
```diff
@@ -142,8 +147,26 @@ readr::write_tsv(filtered_gsva_scores, file.path(results_dir, "atrt_gsva.tsv"))
...
+                   by = c("Tumor_Sample_Barcode" = "Kids_First_Biospecimen_ID")) %>%
   dplyr::filter(sample_id %in% atrt_df$sample_id)
 
 # Write to file
 readr::write_tsv(tmb_df, file.path(results_dir, "atrt_tmb.tsv"))
+
+#### Filter GISTIC data --------------------------------------------------------
+
+gistic_df <- gistic_df %>%
+  as.data.frame() %>%
```
Comment (jaclyn-taroni) [inline analyses/molecular-subtyping-ATRT/00-subset-files-for-ATRT.R]: Do you need to call `as.data.frame()` here to get the `tibble::column_to_rownames` step to work? Asking because you set this as a `data.frame` after the transpose. If it's necessary to do this twice, I think it may be because `gistic_df` is a `data.table` (see my comment above).

File: analyses/molecular-subtyping-ATRT/00-subset-files-for-ATRT.R [hunk 3]
```diff
@@ -142,8 +147,26 @@ readr::write_tsv(filtered_gsva_scores, file.path(results_dir, "atrt_gsva.tsv"))
...
+  tibble::column_to_rownames("Chromosome Arm") %>%
+  t() %>%
+  as.data.frame() %>%
+  tibble::rownames_to_column("Kids_First_Biospecimen_ID") %>%
+  dplyr::left_join(select_metadata, by = "Kids_First_Biospecimen_ID") %>%
+  dplyr::filter(sample_id %in% atrt_df$sample_id) %>%
```
Comment (jaclyn-taroni) [inline analyses/molecular-subtyping-ATRT/00-subset-files-for-ATRT.R]: I think you can move up the filtering step such that you are removing non-ATRT biospecimens from the GISTIC data and then join with the metadata. In general if you can filter before joining, that is a better design pattern because you end up working with smaller `data.frame`.

File: analyses/molecular-subtyping-ATRT/01-ATRT-molecular-subtyping-data-prep.Rmd [hunk 4]
```diff
@@ -168,14 +176,19 @@ metadata <- metadata %>%
     age_at_diagnosis_days,
     germline_sex_estimate,
     primary_site
-  )
+  ) %>%
+  dplyr::group_by(
+    sample_id,
+    Kids_First_Participant_ID
+  ) %>%
+  dplyr::summarize_all(function(x) paste(sort(unique(x)), collapse = ", "))
```
Comment (jaclyn-taroni) [inline analyses/molecular-subtyping-ATRT/01-ATRT-molecular-subtyping-data-prep.Rmd]: @cbethell if we do this collapse biospecimen IDs step up front, we can drop the biospecimen ID columns as we go since we join on `sample_id`. This means we don't have to collapse at the end and run the risk of having a bunch of duplicated columns.

File: analyses/molecular-subtyping-ATRT/00-subset-files-for-ATRT.R [hunk 5]
```diff
@@ -81,26 +81,31 @@ tmb_df <-
...
 
-# Filter metadata for `ATRT` and define `location_summary` based on values in
-# `primary_site`
 atrt_df <- metadata %>%
   dplyr::filter(short_histology == "ATRT",
-                experimental_strategy == "RNA-Seq")
-
-# Write to file
-readr::write_tsv(atrt_df, file.path(results_dir, "atrt_histologies.tsv"))
+                sample_type == "Tumor",
```
Comment (jaclyn-taroni) [inline analyses/molecular-subtyping-ATRT/00-subset-files-for-ATRT.R]: Filtering by `experimental_strategy == "RNA-Seq"` also effectively drops normal samples, but it also made it such that any ATRT samples missing RNA-seq data got dropped. We don't need to write the ATRT subset histologies file as a TSV because the CI histologies file is always the same file as a data release: 

https://github.com/AlexsLemonade/OpenPBTA-analysis/blob/2cbf526e24d92c11f093b33aa31663546b3fbb57/analyses/create-subset-files/create_subset_files.sh#L41

The reason why we can include the full `pbta-histologies.tsv` for testing is two-fold: 1) it's pretty small and 2) having "extra" biospecimens/samples/participants in the histologies file helps test for brittle code.

File: analyses/molecular-subtyping-ATRT/01-ATRT-molecular-subtyping-data-prep.Rmd [hunk 6]
```diff
@@ -52,14 +52,18 @@ if (!dir.exists(results_dir)) {
...
 
+# Read in non-subsetted metadata
+metadata <- readr::read_tsv(file.path(root_dir, "data", "pbta-histologies.tsv"))
+
 # Read in ATRT subset metadata
-metadata <-
-  readr::read_tsv(file.path(input_dir, "atrt_histologies.tsv"))
+subset_metadata <- metadata %>%
+  dplyr::filter(short_histology == "ATRT",
+                sample_type == "Tumor",
```
Comment (jaclyn-taroni) [inline analyses/molecular-subtyping-ATRT/01-ATRT-molecular-subtyping-data-prep.Rmd]: Getting rid of the normal and cell line data here.

File: analyses/molecular-subtyping-ATRT/01-ATRT-molecular-subtyping-data-prep.Rmd [hunk 7]
```diff
@@ -241,50 +254,60 @@ long_stranded_expression <- scale(t(filtered_expression),
...
   head(n = 15)
 
 # Join expression data with metadata filtered for `ATRT`
-atrt_expression_df <- metadata %>%
+atrt_expression_df <- collapsed_metadata %>%
   dplyr::left_join(expression_metadata,
                    by = "sample_id")
+
+# Remove data we no longer need
+rm(filtered_expression, long_stranded_expression, expression_metadata)
```
Comment (jaclyn-taroni) [inline analyses/molecular-subtyping-ATRT/01-ATRT-molecular-subtyping-data-prep.Rmd]: Removing from the workspace as we go makes it such that we don't have `data.frame` we no longer need in memory.

File: analyses/molecular-subtyping-ATRT/01-ATRT-molecular-subtyping-data-prep.Rmd [hunk 8]
```diff
@@ -241,50 +254,60 @@ long_stranded_expression <- scale(t(filtered_expression),
...
-    )), collapse = ", ")
-  )
-
-# Display `cn_metadata`
-cn_df %>%
+  tidyr::spread(gene_symbol, status, fill = "neutral") %>%
+  dplyr::rename(SMARCB1_focal_status = SMARCB1,
+                SMARCA4_focal_status = SMARCA4)
+
+# add in the samples that did not have any copy number changes for either
```
Comment (jaclyn-taroni) [inline analyses/molecular-subtyping-ATRT/01-ATRT-molecular-subtyping-data-prep.Rmd]: Before, if a biospecimen did not have evidence for an alteration in either _SMARCB1_ or _SMARCA4_, it would look like it was "missing" in the final data.frame. This is likely because we filter out neutral events here: 

https://github.com/AlexsLemonade/OpenPBTA-analysis/blob/2cbf526e24d92c11f093b33aa31663546b3fbb57/analyses/focal-cn-file-preparation/01-prepare-cn-file.R#L262

to save on memory for the annotation step and disk space when we save the prepped file. So here, we assume that if a sample appears _anywhere_ in the CNVkit file that we did assay it but that there was no evidence of a copy number change so we mark it neutral.

@jharenza you should be aware of this assumption as you look at the table! Not sure how common it is to essentially 'miss' segments when calling copy number from WGS and therefore _not_ assay  _SMARCB1_ and _SMARCA4_ for those samples...

File: analyses/molecular-subtyping-ATRT/01-ATRT-molecular-subtyping-data-prep.Rmd [hunk 9]
```diff
@@ -329,44 +355,82 @@ tmb_df %>%
...
+rm(tmb_df, atrt_expression_cn_df)
+```
+
+### GISTIC data
+
+```{r}
+gistic_df <- gistic_df %>%
+  dplyr::select(-Kids_First_Biospecimen_ID) %>%
+  dplyr::group_by(sample_id) %>%
+  dplyr::mutate(chr_22q_loss = dplyr::case_when(`22q` == -1 ~ "Yes",
```
Comment (jaclyn-taroni) [inline analyses/molecular-subtyping-ATRT/01-ATRT-molecular-subtyping-data-prep.Rmd]: @jharenza we only have `-1` and `0` values in `22q` for these samples: https://github.com/AlexsLemonade/OpenPBTA-analysis/blob/7f012635616c171531b627606f6607874db04d30/analyses/molecular-subtyping-ATRT/atrt-subset/atrt_gistic_broad_values.tsv

Unless GISTIC itself incorporates the tumor ploidy information from `pbta-histologies.tsv`, we are not using any info about ploidy here. I think given that a `-1` would be a loss even when ploidy is 2, this might be fine but wanted to make you aware.

File: analyses/molecular-subtyping-ATRT/00-subset-files-for-ATRT.R (hunk 2 above)
Comment (cbethell) [reply analyses/molecular-subtyping-ATRT/00-subset-files-for-ATRT.R]: Yes, you are correct.

File: analyses/molecular-subtyping-ATRT/01-ATRT-molecular-subtyping-data-prep.Rmd (hunk 9 above)
Comment (jharenza) [reply analyses/molecular-subtyping-ATRT/01-ATRT-molecular-subtyping-data-prep.Rmd]: yes, GISTIC only uses -2, -1, 0, 1, 2 to my knowledge

File: analyses/molecular-subtyping-ATRT/01-ATRT-molecular-subtyping-data-prep.Rmd (hunk 8 above)
Comment (jharenza) [reply analyses/molecular-subtyping-ATRT/01-ATRT-molecular-subtyping-data-prep.Rmd]: thanks for the heads up! The normal output from segmentation algorithms don't contain neutral segments, so this makes sense.
